src/backend_helper.c

```diff
@@ -65,12 +60,14 @@ char *get_default_printer(BackendObj *b)
         if ((attr = ippFindAttribute(response, "printer-name",
                                      IPP_TAG_NAME)) != NULL)
         {
+            g_free(b->default_printer);
             b->default_printer = g_strdup(ippGetString(attr, 0, NULL));
             ippDelete(response);
             return b->default_printer;
         }
     }
     ippDelete(response);
+    g_free(b->default_printer);
```

Comment:
Do not return the string "NA" when there is no default printer, return an empty string ""instead, as there can be users who have a print queue named "NA", while the empty string is not a legal print queue name, so one clearly sees that no default printer could be determined.

Reply:
I agree that returning "NA" is not a good idea. But that's preexisting, unrelated to this PR.
Can we therefore handle it independently? (I can submit another PR once this one is done.)

Wouldn't using `NULL` instead of an empty string be even clearer? (That's also what `get_new_BackendObj` sets initially.)

Reply:
OK, so please do a separate PR for that.

`NULL` is also OK, no problem and more consistent with the rest of the code.

Reply:
OK, thanks. I'll create the other PR once this one is merged, to avoid merge conflicts.

Reply:
> I'll create the other PR once this one is merged, to avoid merge conflicts.

-> https://github.com/OpenPrinting/cpdb-backend-cups/pull/35